MoneyDataType/Currency.cs

```diff
@@ -0,0 +1,136 @@
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using System;
+using System.Text.Json.Serialization;
+using OrchardCore.Commerce.Abstractions;
+using OrchardCore.Commerce.Serialization;
+
+namespace OrchardCore.Commerce.Money
+{
+    [JsonConverter(typeof(CurrencyConverter))]
+    [Newtonsoft.Json.JsonConverter(typeof(LegacyCurrencyConverter))]
+    [DebuggerDisplay("{DebuggerDisplay,nq}")]
+    public struct Currency : ICurrency
+    {
+        internal readonly CultureInfo _culture;
+
+        public Currency(string cultureCode)
+            :this (CultureInfo.GetCultureInfo(cultureCode))
+        { }
+
+        internal Currency(CultureInfo culture)
+        {
+            if (culture is null)
+                throw new ArgumentNullException(nameof(culture));
+            if (culture.EnglishName.StartsWith("Unknown Locale") || culture.EnglishName.StartsWith("Invariant Language"))
+                throw new ArgumentOutOfRangeException(nameof(culture));
+
+            var region = new RegionInfo(culture.LCID);
+            if (region is null)
+                throw new ArgumentNullException(nameof(region));
+
+
+            EnglishName = region.CurrencyEnglishName;
+            NativeName = region.CurrencyNativeName;
+            Symbol = region.CurrencySymbol;
+            IsoCode = region.ISOCurrencySymbol;
+            DecimalPlaces = culture.NumberFormat.CurrencyDecimalDigits;
+            _culture = culture;
+        }
+
+        public Currency(string englishName, string nativeName, string symbol, string iSOSymbol, int decimalDigits = 2) 
+        {
+            EnglishName = englishName;
+            NativeName = nativeName;
+            Symbol = symbol;
+            IsoCode = iSOSymbol;
+            DecimalPlaces = decimalDigits;
+            _culture = null;
+        }
+
+        public string Culture => _culture?.Name??IsoCode;
+
+        public int DecimalPlaces { get; }
+
+        public string IsoCode { get; }
+
+        public string Symbol { get; }
+
+        public string EnglishName { get; }
```

Comment:
Because English and Native are very different values, and like RegionInfo which provides both, it depends on the implementation as to which is used.  Will revert back to single name if you prefer, but then it should probably use the RegionInfo.DisplayName?

Not sure how this would be overwritten using OrchardCore.Localization though.

Reply:
Of course they are different, but there doesn't seem to be a good reason to provide an English version and not other languages. We could decide to make that fully localizable somehow, or remove English. Let's remove English for now.

Reply:
English name removed, NativeName renamed to Name.  Contains native name now. So this now looks like this:

| Name | IsoCode | Symbol | decimals |
|---|---|---|---|
| টকা | INR | ₹ | 2 |
| Aзәрбајҹан манаты | AZN | ₼ | 2 |
| һум | RUB | ₽ | 2 |
| беларускі рубель | BYN | Br | 2 |
| български лев | BGN | лв. | 2 |
| Nigerian Naira | NGN | ₦ | 2 |
| বাংলাদেশী টাকা | BDT | ৳ | 2 |